browser/ui/webui/brave_shields/shields_panel_ui.cc

```diff
@@ -58,7 +63,6 @@ void ShieldsPanelUI::CreatePanelHandler(
 
   panel_handler_ =
       std::make_unique<ShieldsPanelHandler>(std::move(panel_receiver), this);
-
   data_handler_ = std::make_unique<ShieldsPanelDataHandler>(
-      std::move(data_handler_receiver), this);
+      std::move(data_handler_receiver), this, browser_);
```

Comment:
we can pass tab_strip_model. but I thought using tab strip model here seems unnecessary detail.

Reply:
makes sense

Reply:
either way is fine, I'd actually prefer tab_strip_model because we should only pass what it actually needs, but I'm fine with browser

Reply:
@simonhong re "unnecessary detail" - I think it's best in general to pass the most specific thing it actually needs because it reduces unnecessary dependencies and also in some cases the more specific thing might be easier to access from the caller

Reply:
example: passing PrefService instead of BrowserContext

Reply:
even if you need PrefService and let's say BookmarkService for example, it's better to pass both than to pass BrowserContext because of the dependency issue and also that makes it easier to mock things for testing

Reply:
Ok, thanks for explaination! 👍🏼